src/libs/actions/Workflow.ts

```diff
@@ -183,7 +184,7 @@ function removeApprovalWorkflow(policyID: string, approvalWorkflow: ApprovalWork
     const updatedEmployees = convertApprovalWorkflowToPolicyEmployees({previousEmployeeList, approvalWorkflow, type: CONST.APPROVAL_WORKFLOW.TYPE.REMOVE});
     const updatedEmployeeList = {...previousEmployeeList, ...updatedEmployees};
 
-    const defaultApprover = policy.approver ?? policy.owner;
+    const defaultApprover = getDefaultApprover(policy);
     // If there is more than one workflow, we need to keep the advanced approval mode (first workflow is the default)
     const hasMoreThanOneWorkflow = Object.values(updatedEmployeeList).some((employee) => !!employee.submitsTo && employee.submitsTo !== defaultApprover);
```

Comment:
For this case, there is nothing noticeable from the UI side, but previously, if we have 2 workflows (including the default one), then we deleted the 2nd one, the optimistic `approvalMode` will be ADVANCED, but BE returns BASIC. After this change, the optimistic `approvalMode` will be BASIC.